sdk/go/common/workspace/plugins.go

```diff
@@ -447,24 +447,21 @@ func (source *githubSource) getHTTPResponse(
 		return resp, length, nil
 	}
 
-	// Wrap 403 rate limit errors with a more helpful message.
+	// Wrap 403 rate limit errors with a more helpful message later. This is a rate limiting error only
+	// if x-ratelimit-remaining is 0.
+	// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
 	var downErr *downloadError
-	if !errors.As(err, &downErr) || downErr.code != 403 {
-		// If we see a 401 error and were using a token we'll disable that token and try again
-		if downErr != nil && downErr.code == 401 && source.token != "" {
+	if !errors.As(err, &downErr) || downErr.code != 403 || downErr.header.Get("x-ratelimit-remaining") != "0" {
+		// If we see a 401 error and we're using a token we'll disable that token and try again
+		fmt.Println(err, downErr)
+		if downErr != nil && (downErr.code == 401 || downErr.code == 403) && source.token != "" {
```

Comment:
I have a hard time with the nested if's and figuring out all the paths through here. I think this is right, but maybe we should split this out or comment differently? The comment on line 450 says we're wrapping 403 errors, why are we *not* wrapping a 403 sometimes here?

Reply:
I tried improving the comment here, and split out a function naming rate limit errors specifically.  What do you think?

Reply:
This is great, thanks!